### PR TITLE
[TextFields] Added check on textinput.font set to avoid sideeffects

### PR DESCRIPTION
**Pull request by edisaverio**
And added alignment of the placeholder to the bottom of the textfield. This aligns the baselines of placeholder and text.

### CODE REVIEW

**Comment by edisaverio** [comment]
I signed It!

**Comment by edisaverio** [comment]
Fix for https://github.com/material-components/material-components-ios/issues/4189

**Comment by jverkoey** [comment]
@andrewoverton Are you able to review this?

**Comment by edisaverio** [comment]
Well should anyone need this urgently I forked to have the fix https://github.com/hazam/material-components-ios . Of course will delete as soon as the PR gets into a release

File: components/TextFields/src/private/MDCTextInputCommonFundament.m [hunk 1]
```diff
@@ -876,7 +876,9 @@ - (void)updatePlaceholderAlpha {
 }
 
 - (void)updatePlaceholderPosition {
-  self.placeholderTop.constant = _textInput.textInsets.top;
+  CGFloat diff = self.textInput.font.lineHeight - self.placeholderLabel.font.lineHeight;
```
Comment (randallli) [inline components/TextFields/src/private/MDCTextInputCommonFundament.m:879]: This looks unrelated to the change you described in the PR description. Is this fixing a different bug?
If it would be great to get it as a different PR.
If not, please explain more how the changes are linked together.

**Review by randallli — CHANGES_REQUESTED** [review]

File: components/TextFields/src/private/MDCTextInputCommonFundament.m (hunk 1 above)
Comment (edisaverio) [reply components/TextFields/src/private/MDCTextInputCommonFundament.m:879]: It's an issue within the same setup that reproduced this bug - If I remove this the component would still be unusable. No problem with actually moving this to a new PR, but I wonder when will it be merged if not linked to an open issue.

Comment (edisaverio) [reply components/TextFields/src/private/MDCTextInputCommonFundament.m:879]: To be precise, the additional fix makes sure that placeholder text and texfield content are bottom aligned, instead of top aligned. The issue is only evident when point sizes of placeholder text and textfield are very different.

**Comment by jverkoey** [comment]
Hey @hazam, thanks for the PR! As @randallli pointed out we do encourage PRs being focused on a single idea at a time.

The reasoning behind this is that it sometimes — even though a PR passes public CI — we may detect a regression in a client app during our release process. When this happens, [our policy](https://github.com/material-components/material-components-ios/blob/develop/contributing/releasing.md#a-note-on-the-role-of-the-release-engineer) is to immediately roll back the commit that introduced the regression and to then proceed with the release. If the regressing commit includes multiple ideas, this unfortunately means all of those ideas will have to be reverted.

Please break this PR up in to two separate PRs. It sounds like you may already have the description written for the new PR:

> the additional fix makes sure that placeholder text and texfield content are bottom aligned, instead of top aligned. The issue is only evident when point sizes of placeholder text and textfield are very different.

**Review by jverkoey — CHANGES_REQUESTED** [review]
Requesting the PR be split in two.

**Comment by edisaverio** [comment]
Never mind, I'm dropping the PR.
Don't have time to wait for 30 days turnaround time for a 4 LOC fix.
